pkg/apis/events/v1alpha1/cloudbuildsource_types.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	duckv1alpha1 "github.com/google/knative-gcp/pkg/apis/duck/v1alpha1"
+	kngcpduck "github.com/google/knative-gcp/pkg/duck"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis/duck"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/webhook/resourcesemantics"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudBuildSource is a specification for a CloudBuildSource resource
+// +k8s:openapi-gen=true
+type CloudBuildSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CloudBuildSourceSpec   `json:"spec,omitempty"`
+	Status CloudBuildSourceStatus `json:"status,omitempty"`
+}
+
+var (
+	_ kmeta.OwnerRefable           = (*CloudBuildSource)(nil)
+	_ resourcesemantics.GenericCRD = (*CloudBuildSource)(nil)
+	_ kngcpduck.PubSubable         = (*CloudBuildSource)(nil)
+	_                              = duck.VerifyType(&CloudBuildSource{}, &duckv1.Conditions{})
+)
+
+// CloudBuildSourceSpec defines the desired state of the CloudBuildSource.
+type CloudBuildSourceSpec struct {
+	// This brings in the PubSub based Source Specs. Includes:
+	// Sink, CloudEventOverrides, Secret, PubSubSecret, and Project
```

Comment:
just removed that PubSubSecret... can you remove it from the comment, and probably all other sources as well? Maybe do a ctrl+f to see where else we have it